service/svc.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+ROOTPROJECTPATH="$(
+    cd -- "$(dirname "$0")/.." >/dev/null 2>&1
+    pwd -P
+)"
+
+function install {
+    path=$1
+    if [ -z "$1" ]; then
+        path="/lib/systemd/system"
+    fi
+    if [ -e "$path/threeal-bot.service" ]; then
+        echo "Service is already installed"
+        exit 1
+    fi
+    go_path=$(which go 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        echo "Go is not installed, please install go (https://go.dev/doc/install)"
+        exit 1
+    fi
+    workdir=$ROOTPROJECTPATH
+    user=$(logname)
+    echo "Installing service..."
+    sed -e "s@<user>@$user@g" -e "s@<workdir>@$workdir@g" -e "s@<goabsolutepath>@$go_path@g" service/threeal-bot.service >/tmp/threeal-bot.service
+    (mv /tmp/threeal-bot.service $path && echo "Done installing service") || (rm /tmp/threeal-bot.service && exit 1)
+}
+
+function uninstall {
+    path=$1
+    if [ -z "$1" ]; then
+        path="/lib/systemd/system/threeal-bot.service"
+    fi
+    if [ ! -e "$path" ]; then
+        echo "Service is not installed"
+        exit 1
+    fi
+    echo "Uninstalling service..."
+    (rm $path && echo "Done uninstalling service") || (exit 1)
```

Comment:
You can use `set -e` instead to handle this. see [this](https://linuxhint.com/bash-set-e/).
In short, that specifier on the top of this script could cause exit if any command return with non zero status.

```
rm $path
echo "Done uninstalling service"
```

That line will exit immediately if `rm` command is failed.